rustls/src/client/tls12.rs

```diff
@@ -405,7 +405,7 @@ impl State<ClientConnectionData> for ExpectServerKx {
             .ok_or_else(|| {
                 cx.common
                     .send_fatal_alert(AlertDescription::DecodeError);
-                Error::CorruptMessagePayload(ContentType::Handshake)
+                Error::CorruptMessage(InvalidMessage::MissingKeyExchange)
```

Comment:
Similar to the earlier PRs, it would good to add some `From` impls to make this slightly less wordy. For example, if there was an `impl From<InvalidMessage> for Error` then we could probably just yield `InvalidMessage::MissingKeyExchange` here, and let the `?` operator take care of wrapping it in `Error`?

Reply:
I modified all of the inline/chained occurrences of this pattern. I left the ones that had just error construction in an un-nested line alone for clarity.